Public/Src/Engine/Processes/SandboxedProcessInfo.cs

```diff
@@ -364,5 +366,451 @@ public byte[] GetUnicodeEnvironmentBlock()
+                    MaxLengthInMemory = maxLengthInMemory,
+                    Timeout = timeout,
+                    NestedProcessTerminationTimeout = nestedProcessTerminationTimeout,
+                    PipSemiStableHash = pipSemiStableHash,
+                    TimeoutDumpDirectory = timeoutDumpDirectory,
+                    SandboxKind = sandboxKind,
+                    PipDescription = pipDescription,
+                    SandboxedProcessStandardFiles = sandboxedProcessStandardFiles,
+                    StandardInputSourceInfo = standardInputSourceInfo,
+                    StandardObserverDescriptor = standardObserverDescriptor
+                };
+            }
+        }
+
+        #endregion
+
+        #region Extra data structure for serialization
+
+        /// <summary>
+        /// Info about the source of standard input.
+        /// </summary>
+        public class StandardInputInfo : IEquatable<StandardInputInfo>
+        {
+            /// <summary>
+            /// File path.
+            /// </summary>
+            public string File { get; }
+
+            /// <summary>
+            /// Raw data.
+            /// </summary>
+            public string Data { get; }
+
+            private StandardInputInfo(string file, string data)
+            {
+                Contract.Requires((file != null && data == null) || (file == null && data != null));
+
+                File = file;
+                Data = data;
+            }
+
+            /// <summary>
+            /// Creates a standard input info where the source comes from a file.
+            /// </summary>
+            public static StandardInputInfo CreateForFile(string file)
+            {
+                Contract.Requires(!string.IsNullOrEmpty(file));
+
+                return new StandardInputInfo(file, null);
+            }
+
+            /// <summary>
+            /// Creates a standard input info where the source comes from raw data.
+            /// </summary>
+            public static StandardInputInfo CreateForData(string data)
+            {
+                Contract.Requires(data != null);
+
+                return new StandardInputInfo(null, data);
+            }
+
+            /// <summary>
+            /// Serializes this instance to a given <paramref name="writer"/>.
+            /// </summary>
+            public void Serialize(BuildXLWriter writer)
+            {
+                Contract.Requires(writer != null);
+
+                writer.WriteNullableString(File);
+                writer.WriteNullableString(Data);
+            }
+
+            /// <summary>
+            /// Deserializes an instance of <see cref="StandardInputInfo"/>
+            /// </summary>
+            public static StandardInputInfo Deserialize(BuildXLReader reader)
+            {
+                Contract.Requires(reader != null);
+
+                return new StandardInputInfo(reader.ReadNullableString(), reader.ReadNullableString());

```

Comment:
named arguments

Reply:
```
tandardInputInfo(
    file: reader.ReadNullableString(), 
    data: reader.ReadNullableString()
);
```

---
In reply to: [276819630](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276819630) [](ancestors = 276819630)